---
title: "DESeq with RRHO and GO analysis"
output: html_notebook
---

This notebook contains the primary DESeq2 and RRHO bulk RNAseq analyses. For both sections, GO analysis is performed on both DEGs/quadrant genes and intersections of those gene lists generated by the Fisher's Exact Test. Additionally, gene lists are exported for vizualiation in Morpheus, an online tool provided by the Broad institue. 


This is an [R Markdown](http://rmarkdown.rstudio.com) Notebook. When you execute code within the notebook, the results appear beneath the code. 

#load libraries 
```{r}
library(tidyverse)
library(ggplot2)
library(ggfortify)
library(ggrepel)
library(GGally)
library(wesanderson)
library(viridis)
library(readxl)
library(car)
library(gridExtra)
library(ez)
library(ggpubr)
library(DESeq2)
library(EnhancedVolcano)
library(RRHO2)
library(DOSE)
library(enrichplot)
library(ggpubr)
library(org.Mm.eg.db) #mouse genome data
library(clusterProfiler)
library(UpSetR)
library(scales)
library(heatmap3)
library(pheatmap)
library(stringi)
library(SuperExactTest)
source("{dir}/Functions/convert_voleToMouse_geneIDs.R")
source("{dir}/Functions/plot_GOanlaysis.R")
```


#import sequencing data for inputs and column data 
```{r}
metrics <- read_excel("{dir}/Input_RNAseq_metrics.xlsx", sheet = "Correct order prelim")
metrics <- mutate_at(metrics, vars("Animal", "Cohort", "Pairing", "Timepoint", "Separation", "ProcessingGroup", "PPrefCat", "DissectionOrder", "CageChange", "ParentCode", "PairingAgeCat", "SACAgeCat", "RIN", "InputRNACat", "TRAP", "InputRNA"), as.factor)
#remove animal 2945 that had a ppref <50%
metrics <- metrics %>% slice(-c(12))

inputs <- read.csv("{dir}/Merged all_inputs.txt", stringsAsFactors=FALSE)
inputs <- inputs %>% filter_all(all_vars(.>10)) %>% distinct(genes, .keep_all = TRUE) %>% filter(!is.na(genes)) %>% column_to_rownames("genes")
#remove animal 2945
inputs <- inputs %>% select(-c(12))
```

#run DESeq
```{r}
dds <- DESeqDataSetFromMatrix(countData = inputs, colData = metrics, design = ~ParentCode + PairingAgeCat + Cohort)
dds <- DESeq(dds)

res.48P <- results(dds, contrast = c("Cohort", "OS48P", "SS48P"), alpha = 0.05, pAdjustMethod = "fdr")
res.48I <- results(dds, contrast = c("Cohort", "OS48I", "SS48I"), alpha = 0.05, pAdjustMethod = "fdr")
res.4P <- results(dds, contrast = c("Cohort", "OS4P", "SS4P"), alpha = 0.05, pAdjustMethod = "fdr")
res.4I <- results(dds, contrast = c("Cohort", "OS4I", "SS4I"), alpha = 0.05, pAdjustMethod = "fdr")

dds.collapsed <- DESeqDataSetFromMatrix(countData = inputs, colData = metrics, design = ~ParentCode + PairingAgeCat + RPCollapsed)
dds.collapsed <- DESeq(dds.collapsed)

res.collapsed <- results(dds.collapsed, contrast = c("RPCollapsed", "OSP", "SSP"), alpha = 0.05, pAdjustMethod = "fdr")

res.48P.df <- as.data.frame(res.48P) %>% rownames_to_column("genes")
res.48I.df <- as.data.frame(res.48I) %>% rownames_to_column("genes")
res.4P.df <- as.data.frame(res.4P) %>% rownames_to_column("genes")
res.4I.df <- as.data.frame(res.4I) %>% rownames_to_column("genes")
res.collpased.df <- as.data.frame(res.collapsed) %>% rownames_to_column("genes")
```
volcano plots
```{r}
EnhancedVolcano(res.48P, lab = NA, x = "log2FoldChange", y = "pvalue", FCcutoff = 0.30, pCutoff = 0.05, col = c("gray70", "gray70", "gray70", "#F21A00"), title = NULL, subtitle = NULL, caption = NULL, border = "full", borderColour = "black", pointSize = 1, cutoffLineWidth = 0.25, labSize = 2, legendPosition = "none", gridlines.major = FALSE, gridlines.minor = FALSE, colAlpha = 0.75) + theme(axis.title.x = element_text(size =10), axis.text.x = element_text(size = 8), axis.title.y = element_text(size =10), axis.text.y = element_text(size =8)) + ylim(c(0,10)) + xlim(c(-4,4))
ggsave("OSvSS 48hrs remain paired.pdf", width = 4, height = 2.5)

EnhancedVolcano(res.48I, lab = NA, x = "log2FoldChange", y = "pvalue", FCcutoff = 0.30, pCutoff = 0.05, col = c("gray70", "gray70", "gray70", "#F21A00"), title = NULL, subtitle = NULL, caption = NULL, border = "full", borderColour = "black", pointSize = 1, cutoffLineWidth = 0.25, labSize = 2, legendPosition = "none", gridlines.major = FALSE, gridlines.minor = FALSE, colAlpha = 0.75) + theme(axis.title.x = element_text(size =10), axis.text.x = element_text(size = 8), axis.title.y = element_text(size =10), axis.text.y = element_text(size =8)) + ylim(c(0,10)) + xlim(c(-4,4))
ggsave("OSvSS 48hrs isolated.pdf", width = 4, height = 2.5)

EnhancedVolcano(res.4P, lab = NA, x = "log2FoldChange", y = "pvalue", FCcutoff = 0.30, pCutoff = 0.05, col = c("gray70", "gray70", "gray70", "#F21A00"), title = NULL, subtitle = NULL, caption = NULL, border = "full", borderColour = "black", pointSize = 1, cutoffLineWidth = 0.25, labSize = 2, legendPosition = "none", gridlines.major = FALSE, gridlines.minor = FALSE, colAlpha = 0.75) + theme(axis.title.x = element_text(size =10), axis.text.x = element_text(size = 8), axis.title.y = element_text(size =10), axis.text.y = element_text(size =8)) + ylim(c(0,10)) + xlim(c(-4,4))
ggsave("OSvSS 4wks paired.pdf", width = 4, height = 2.5)

EnhancedVolcano(res.4I, lab = NA, x = "log2FoldChange", y = "pvalue", FCcutoff = 0.30, pCutoff = 0.05, col = c("gray70", "gray70", "gray70", "#F21A00"), title = NULL, subtitle = NULL, caption = NULL, border = "full", borderColour = "black", pointSize = 1, cutoffLineWidth = 0.25, labSize = 2, legendPosition = "none", gridlines.major = FALSE, gridlines.minor = FALSE, colAlpha = 0.75) + theme(axis.title.x = element_text(size =10), axis.text.x = element_text(size = 8), axis.title.y = element_text(size =10), axis.text.y = element_text(size =8)) + ylim(c(0,10)) + xlim(c(-4,4))
ggsave("OSvSS 4wks isolated.pdf", width = 4, height = 2.5)

EnhancedVolcano(res.collapsed, lab = NA, x = "log2FoldChange", y = "pvalue", FCcutoff = 0.30, pCutoff = 0.05, col = c("gray70", "gray70", "gray70", "#F21A00"), title = NULL, subtitle = NULL, caption = NULL, border = "full", borderColour = "black", pointSize = 1, cutoffLineWidth = 0.25, labSize = 2, legendPosition = "none", gridlines.major = FALSE, gridlines.minor = FALSE, colAlpha = 0.75) + theme(axis.title.x = element_text(size =10), axis.text.x = element_text(size = 8), axis.title.y = element_text(size =10), axis.text.y = element_text(size =8)) + ylim(c(0,10)) + xlim(c(-4,4))
ggsave("OSvSS collapsed pair bond.pdf", width = 4, height = 2.5)
```

make dfs of the up and downregulated genes for each DESeq results df
```{r}
degs.48P <- deg(res.48P.df)
degs.48I <- deg(res.48I.df)
degs.4P <- deg(res.4P.df)
degs.4I <- deg(res.4I.df)
degs.collapsed <- deg(res.collpased.df)
```

Fisher's extact test set up for Upset plots
```{r}
#total number of genes
total = 12128

#graphing aesthetics
paletteFunc <- colorRampPalette(c('white', 'grey35'));
palette     <- paletteFunc(30)
pvalueShades = palette
```

make lists for FET
```{r}
#make lists for up DEGs and down DEGs for 48P and 4P 
upset_list <- list(degs.48P[["upreg_DEGs_genesonly"]]$genes, degs.48P[["downreg_DEGs_genesonly"]]$genes, degs.4P[["upreg_DEGs_genesonly"]]$genes, degs.4P[["downreg_DEGs_genesonly"]]$genes)
names(upset_list) <- c("Up DEGs 48P", "Down DEGs 48P", "Up DEGs 4P", "Down DEGs 4P")

#make lists for all DEGs, up DEGS, and down DEGS for collapsed PBonds and isolated DESeqs
upset_list2 <- list(degs.collapsed[["up.downregulated_DEGs_genesonly"]]$genes, degs.48I[["up.downregulated_DEGs_genesonly"]]$genes, degs.4I[["up.downregulated_DEGs_genesonly"]]$genes)
names(upset_list2) <- c("Up/down DEGs collasped PBond", "Up/down DEGs 48I", "Up/down DEGS 4I")

upset_up2 <- list(degs.collapsed[["upreg_DEGs_genesonly"]]$genes, degs.48I[["upreg_DEGs_genesonly"]]$genes, degs.4I[["upreg_DEGs_genesonly"]]$genes)
names(upset_up2) <- c("Up DEGS collapsed PBond", "Up DEGS 48I", "Up DEGS 4I")

upset_down2 <- list(degs.collapsed[["downreg_DEGs_genesonly"]]$genes, degs.48I[["downreg_DEGs_genesonly"]]$genes, degs.4I[["downreg_DEGs_genesonly"]]$genes)
names(upset_down2) <- c("Down DEGs collapsed PBond", "Down DEGS 48I", "Down DEGS 4I")
```

run Fisher's Exact Tests
```{r}
res <- supertest(upset_list, n = total)
res_all2 <- supertest(upset_list2, n = total)
res_up2 <- supertest(upset_up2, n = total)
res_down2 <- supertest(upset_down2, n = total)
```

plot FETs
```{r}
plot(res, Layout="landscape", sort.by="size", margin=c(0.5,5,1,2), keep.empty.intersections = FALSE, show.expected.overlap=TRUE,expected.overlap.style="box", color.expected.overlap='black', color.on = "black", color.off = "whitesmoke", heatmapColor = pvalueShades, bar.split=c(120,1040))

plot(res_all2, Layout="landscape", sort.by="size", margin=c(0.5,5,1,2), keep.empty.intersections = FALSE, show.expected.overlap=TRUE,expected.overlap.style="box", color.expected.overlap='black', color.on = "black", color.off = "whitesmoke", heatmapColor = pvalueShades, bar.split=c(100,1195))

plot(res_up2, Layout="landscape", sort.by="size", margin=c(0.5,5,1,2), keep.empty.intersections = FALSE, show.expected.overlap=TRUE,expected.overlap.style="box", color.expected.overlap='black', color.on = "black", color.off = "whitesmoke", heatmapColor = pvalueShades, bar.split=c(100,870))

plot(res_down2, Layout="landscape", sort.by="size", margin=c(0.5,5,1,2), keep.empty.intersections = FALSE, show.expected.overlap=TRUE, expected.overlap.style="box", color.expected.overlap='black', color.on = "black", color.off = "whitesmoke", heatmapColor = pvalueShades, bar.split = c(10, 295))
```

getting overlapping gene lists from Upset plots of DESeq data
```{r}
res_up2_colPBond <- as.data.frame(res_up2[["x"]][["Up DEGS collapsed PBond"]])
res_up2_48I <- as.data.frame(res_up2[["x"]][["Up DEGS 48I"]])

true_false <- as.data.frame(res_up2[["x"]][["Up DEGS collapsed PBond"]] %in% res_up2[["x"]][["Up DEGS 48I"]])
overlap_colPBond_48I <- cbind(res_up2_colPBond, true_false) %>% dplyr::rename("colPbond_genes" = 1) %>% dplyr::rename("true_false" = 2)
overlap_colPBond_48I <- overlap_colPBond_48I %>% filter(true_false == "TRUE")
write_csv(overlap_colPBond_48I, "overlap in upreg DEGs between colPBond and 48I.txt")

true_false <- as.data.frame(res_up2[["x"]][["Up DEGS collapsed PBond"]] %in% res_up2[["x"]][["Up DEGS 4I"]])
overlap_colPBond_4I <- cbind(res_up2_colPBond, true_false) %>% dplyr::rename("colPbond_genes" = 1) %>% dplyr::rename("true_false" = 2)
overlap_colPBond_4I <- overlap_colPBond_4I %>% filter(true_false == "TRUE")
write_csv(overlap_colPBond_4I, "overlap in upreg DEGs between colPBond and 4I.txt")

true_false <- as.data.frame(res_up2[["x"]][["Up DEGS 48I"]] %in% res_up2[["x"]][["Up DEGS 4I"]])
overlap_colPBond_seps <- cbind(res_up2_48I, true_false) %>% dplyr::rename("sep48_genes" = 1) %>% dplyr::rename("true_false" = 2)
overlap_colPBond_seps <- overlap_colPBond_seps %>% filter(true_false == "TRUE")
write_csv(overlap_colPBond_seps, "overlap in upreg DEGs between 48I and 4I.txt")
```

I want to use this tool (https://software.broadinstitute.org/morpheus/) to compare the transcriptional signatures between ST and LT remain paired groups.
make dfs that have the following columns: genes, L2FC ST paired, L2FC LT paired 
```{r}
#for each individual DESeq
final_RPmorpheus <- morpheus(res.48P.df, res.4P.df)
final_sepmorpheus <- morpheus(res.48I.df, res.4I.df)
final_all <- left_join(final_RPmorpheus, final_sepmorpheus, by = "genes")

#for collapsed PBond vs ST and LT sep DESeqs
morpheus_colPB <- res.collpased.df[,c(1,3)] %>% dplyr::rename("L2FC.colPBond" = 2)
morpheus_STsep <- res.48I.df[,c(1,3)] %>% dplyr::rename("L2FC.STsep" = 2)
morpheus_LTsep <- res.4I.df[,c(1,3)] %>% dplyr::rename("L2FC.LTsep" = 2)
final_morpheus_colPB <- left_join(morpheus_colPB, morpheus_STsep, by = "genes")
final_morpheus_colPB <- left_join(final_morpheus_colPB, morpheus_LTsep, by = "genes")

#save csv files
write_csv(final_RPmorpheus, "morpheus_48Pref_4Pcomp.txt")
write_csv(final_sepmorpheus, "morpheus_48Iref_4Icomp.txt")
write_csv(final_all, "morpheus_allGroups_order48P_4P_48I_4I.txt")
write_csv(final_morpheus_colPB, "morpheus_colPBvsSeparations.txt")
```

GO analysis for DESeq DEGs
import mouse genome for converting ids
```{r}
mart_export_PVEnsembl_mouseEnsembl_mousegeneName <- read.csv("{dir}/mart_export_PVEnsembl_mouseEnsembl_mousegeneName.txt", stringsAsFactors=FALSE)

#getting a biomart df that only contains rows that have mouse stable IDs 
biomart_mouseE <- filter(mart_export_PVEnsembl_mouseEnsembl_mousegeneName, trimws(Mouse.gene.stable.ID) !="")
```

#make dfs of just genes for GO analysis
```{r}
go.48P.up.degs <- as.data.frame(degs.48P$upreg_DEGs_genesonly)
go.48P.down.degs <- as.data.frame(degs.48P$downreg_DEGs_genesonly)
go.4P.up.degs <- as.data.frame(degs.4P$upreg_DEGs_genesonly)
go.4P.down.degs <- as.data.frame(degs.4P$downreg_DEGs_genesonly)

go.48I.up.degs <- as.data.frame(degs.48I$upreg_DEGs_genesonly)
go.48I.down.degs <- as.data.frame(degs.48I$downreg_DEGs_genesonly)
go.4I.up.degs <- as.data.frame(degs.4I$upreg_DEGs_genesonly)
go.4I.down.degs <- as.data.frame(degs.4I$downreg_DEGs_genesonly)

go.collapsed.up.degs <- as.data.frame(degs.collapsed$upreg_DEGs_genesonly)
go.collapsed.down.degs <- as.data.frame(degs.collapsed$downreg_DEGs_genesonly)
```

#convert vole to mouse genes for GO
```{r}
go.48P.up.degs <- vole_to_mouse(go.48P.up.degs)
go.48P.down.degs <- vole_to_mouse(go.48P.down.degs)
go.4P.up.degs <- vole_to_mouse(go.4P.up.degs)
go.4P.down.degs <- vole_to_mouse(go.4P.down.degs)

go.48I.up.degs <- vole_to_mouse(go.48I.up.degs)
go.48I.down.degs <- vole_to_mouse(go.48I.down.degs)
go.4I.up.degs <- vole_to_mouse(go.4I.up.degs)
go.4I.down.degs <- vole_to_mouse(go.4I.down.degs)

go.collapsed.up.degs <- vole_to_mouse(go.collapsed.up.degs)
go.collapsed.down.degs <- vole_to_mouse(go.collapsed.down.degs)
```

make GO term dotplots for Remain paired cohorts
```{r}
#I previously determined that Biological Processes is the best category for GO
go.48P.up.plot.bp <- enrichGO(go.48P.up.degs$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "BP", pAdjustMethod = "fdr", readable = TRUE)
go.48P.down.plot.bp <- enrichGO(go.48P.down.degs$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "BP", pAdjustMethod = "fdr", readable = TRUE)
go.4P.up.plot.bp <- enrichGO(go.4P.up.degs$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "BP", pAdjustMethod = "fdr", readable = TRUE)
go.4P.down.plot.bp <- enrichGO(go.4P.down.degs$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "BP", pAdjustMethod = "fdr", readable = TRUE)

#plot
barplot(go.48P.up.plot.bp, showCategory =  20)
barplot(go.48P.down.plot.bp, showCategory = 20)
barplot(go.4P.up.plot.bp, showCategory = 20)
barplot(go.4P.down.plot.bp, showCategory = 20)

#define df of selected GO terms
forplot.go.48P.up <- go.48P.up.plot.bp[c(1,3,4,5,8,9,11,15,16,17),]
forplot.go.48P.down <- go.48P.down.plot.bp[c(1,2,4,11,12,14,15,18,21,23),]
forplot.go.4P.up <- go.4P.up.plot.bp[c(1:6,8,9,14,16),]
forplot.go.4P.down <- go.4P.down.plot.bp[c(1:3,7:13),]

#plot and save plots
```


make GO term plots for separated cohorts
```{r}
#I previously determined that Biological Processes is the best category for GO
go.48I.up.plot.bp <- enrichGO(go.48I.up.degs$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "BP", pAdjustMethod = "fdr", readable = TRUE)
go.48I.down.plot.bp <- enrichGO(go.48I.down.degs$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "BP", pAdjustMethod = "fdr", readable = TRUE)
go.4I.up.plot.bp <- enrichGO(go.4I.up.degs$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "BP", pAdjustMethod = "fdr", readable = TRUE)
go.4I.down.plot.bp <- enrichGO(go.4I.down.degs$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "BP", pAdjustMethod = "fdr", readable = TRUE)

#plot
barplot(go.48I.up.plot.bp, showCategory =  20)
barplot(go.48I.down.plot.bp, showCategory = 20)
barplot(go.4I.up.plot.bp, showCategory = 20)
barplot(go.4I.down.plot.bp, showCategory = 20)

#define df of selected GO terms
forplot.go.48I.up <- go.48I.up.plot.bp[c(1:5,12,13,15,17,20),]
forplot.go.48I.down <- go.48I.down.plot.bp[c(1:9,11),]
forplot.go.4I.up <- go.4I.up.plot.bp[c(1,3,5,7,9,10,13,16,18,20),]
forplot.go.4I.down <- go.4I.down.plot.bp[c(1:10),]

#plot and save plots
```

make GO term dotplots for the collapsed PBond degs
```{r}
#I previously determined that Biological Processes is the best category for GO
go.colPB.up.plot.bp <- enrichGO(go.collapsed.up.degs$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "BP", pAdjustMethod = "fdr", readable = TRUE)
go.colPB.down.plot.bp <- enrichGO(go.collapsed.down.degs$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "BP", pAdjustMethod = "fdr", readable = TRUE)

#plot
barplot(go.colPB.up.plot.bp, showCategory =  20)
barplot(go.colPB.down.plot.bp, showCategory = 20)

#define df of selected GO terms
forplot.go.colPB.up <- go.colPB.up.plot.bp[c(2,3,7,9,12,15,16,18,34,58),]
forplot.go.colPB.down <- go.colPB.down.plot.bp[c(4, 6, 7:9, 13:15, 31, 76),]

#plot and save
```




RRHO
make RRHO dfs
```{r}
OSvSS48I <- res.48I.df[, c(1,3,6)] %>% mutate(rankValue = -1 * log10(pvalue) * sign(log2FoldChange)) %>% arrange(desc(rankValue))
OSvSS48P <- res.48P.df[, c (1,3,6)] %>% mutate(rankValue = -1 * log10(pvalue) * sign(log2FoldChange)) %>% arrange(desc(rankValue))
OSvSS4I <- res.4I.df[, c(1,3,6)] %>% mutate(rankValue = -1 * log10(pvalue) * sign(log2FoldChange)) %>% arrange(desc(rankValue))
OSvSS4P <- res.4P.df[, c(1,3,6)] %>% mutate(rankValue = -1 * log10(pvalue) * sign(log2FoldChange)) %>% arrange(desc(rankValue))
PBcol <- res.collpased.df[, c(1,3,6)] %>% mutate(rankValue = -1 * log10(pvalue) * sign(log2FoldChange)) %>% arrange(desc(rankValue))

#make dfs that only have gene name and rankValue
OSvSS48I <- OSvSS48I[,c(1,4)]
OSvSS48P <- OSvSS48P[,c(1,4)]
OSvSS4I <- OSvSS4I[,c(1,4)]
OSvSS4P <- OSvSS4P[,c(1,4)]
PBcol <- PBcol[,c(1,4)]
```

RRHO2 for each comparison that include the pvalue and fisher plots
```{r}
OSvSS.paired.pvalue <- RRHO2_initialize(OSvSS4P, OSvSS48P, labels = c("OS vs SS Paired at 4wks", "OS vs SS Paired at 48hrs"), stepsize = 100, boundary = 0.02)
OSvSS.paired.fisher <- RRHO2_initialize(OSvSS4P, OSvSS48P, labels = c("Long-term pair bond", "Short-term pair bond"), stepsize = 100, boundary = 0.02, method = "fisher")

OSvSS.col.ST <- RRHO2_initialize(PBcol, OSvSS48I, labels = c("OS vs SS collasped pair bond", "OS vs SS Short-term separation"), stepsize = 100, boundary = 0.02)
OSvSS.col.ST.fisher <- RRHO2_initialize(PBcol, OSvSS48I, labels = c("Combined pair bond", "Short-term partner separation"), stepsize = 100, boundary = 0.02, method = "fisher")

OSvSS.col.LT <- RRHO2_initialize(PBcol, OSvSS4I, labels = c("OS vs SS collasped pair bond", "OS vs SS long-term separation"), stepsize = 100, boundary = 0.02)
OSvSS.col.LT.fisher <- RRHO2_initialize(PBcol, OSvSS4I, labels = c("Combined pair bond", "Long-term partner separation"), stepsize = 100, boundary = 0.02, method = "fisher")

OSvSS.ST.LT <- RRHO2_initialize(OSvSS48I, OSvSS4I, labels = c("OS v SS short-term separation", "OS v SS long-term separation"), stepsize = 100, boundary = 0.02)

OSvSS.LT.ST <- RRHO2_initialize(OSvSS4I, OSvSS48I, labels = c("OS v SS long-term separation", "OS v SS short-term separation"), stepsize = 100, boundary = 0.02)

#plots
pdf("OSvSS.paired.pvalue_scaled.pdf", width = 6.5, height = 5, onefile = TRUE)
RRHO2_heatmap(OSvSS.paired.pvalue, maximum = 300, minimum = 1, colorGradient = viridis(101, option = "A"))
dev.off()
pdf("OSvSS.col.ST.pvalue_scaled.pdf", width = 6.5, height = 5, onefile = TRUE)
RRHO2_heatmap(OSvSS.col.ST, maximum = 300, minimum = 1, colorGradient = viridis(101, option = "A"))
dev.off()
pdf("OSvSS.col.LT.pvalue_scaled.pdf", width = 6.5, height = 5, onefile = TRUE)
RRHO2_heatmap(OSvSS.col.LT, maximum = 300, minimum = 1, colorGradient = viridis(101, option = "A"))
dev.off()
pdf("OSvSS.ST.LT.pvalue_scaled.pdf", width = 6.5, height = 5, onefile = TRUE)
RRHO2_heatmap(OSvSS.ST.LT, maximum = 300, minimum = 1, colorGradient = viridis(101, option = "A"))
dev.off()
pdf("OSvSS.paired.fisher_scaled.pdf", width = 6.5, height = 5, onefile = TRUE)
RRHO2_heatmap(OSvSS.paired.fisher, maximum = 6, colorGradient = viridis(101, option = "A"))
dev.off()
pdf("OSvSS.col.ST.fisher_scaled.pdf", width = 6.5, height = 5, onefile = TRUE)
RRHO2_heatmap(OSvSS.col.ST.fisher, maximum = 6, colorGradient = viridis(101, option = "A"))
dev.off()
pdf("OSvSS.col.LT.fisher_scaled.pdf", width = 6.5, height = 5, onefile = TRUE)
RRHO2_heatmap(OSvSS.col.LT.fisher, maximum = 6, colorGradient = viridis(101, option = "A"))
dev.off()
```

make and save dfs of all gene lists from all quadrants of the three RRHOs
```{r}
#OSvSS.paired.pvalue
paired.uu <- as.data.frame(stri_list2matrix(OSvSS.paired.pvalue[["genelist_uu"]])) %>% rename("list1_uu" = 1, "list2_uu" = 2, "overlap_uu" = 3)
paired.dd <- as.data.frame(stri_list2matrix(OSvSS.paired.pvalue[["genelist_dd"]])) %>% rename("list1_dd" = 1, "list2_dd" = 2, "overlap_dd" = 3)
paired.du <- as.data.frame(stri_list2matrix(OSvSS.paired.pvalue[["genelist_du"]])) %>% rename("list1_du" = 1, "list2_du" = 2, "overlap_du" = 3)
paired.ud <- as.data.frame(stri_list2matrix(OSvSS.paired.pvalue[["genelist_ud"]])) %>% rename("list1_ud" = 1, "list2_ud" = 2, "overlap_ud" = 3)

#OSvSS.col.ST
st.uu <- as.data.frame(stri_list2matrix(OSvSS.col.ST[["genelist_uu"]])) %>% rename("list1_uu" = 1, "list2_uu" = 2, "overlap_uu" = 3)
st.dd <- as.data.frame(stri_list2matrix(OSvSS.col.ST[["genelist_dd"]])) %>% rename("list1_dd" = 1, "list2_dd" = 2, "overlap_dd" = 3)
st.du <- as.data.frame(stri_list2matrix(OSvSS.col.ST[["genelist_du"]])) %>% rename("list1_du" = 1, "list2_du" = 2, "overlap_du" = 3)
st.ud <- as.data.frame(stri_list2matrix(OSvSS.col.ST[["genelist_ud"]])) %>% rename("list1_ud" = 1, "list2_ud" = 2, "overlap_ud" = 3)

#OSvSS.col.LT
lt.uu <- as.data.frame(stri_list2matrix(OSvSS.col.LT[["genelist_uu"]])) %>% rename("list1_uu" = 1, "list2_uu" = 2, "overlap_uu" = 3)
lt.dd <- as.data.frame(stri_list2matrix(OSvSS.col.LT[["genelist_dd"]])) %>% rename("list1_dd" = 1, "list2_dd" = 2, "overlap_dd" = 3)
lt.du <- as.data.frame(stri_list2matrix(OSvSS.col.LT[["genelist_du"]])) %>% rename("list1_du" = 1, "list2_du" = 2, "overlap_du" = 3)
lt.ud <- as.data.frame(stri_list2matrix(OSvSS.col.LT[["genelist_ud"]])) %>% rename("list1_ud" = 1, "list2_ud" = 2, "overlap_ud" = 3)

#OSvSS.ST.LT
stlt.uu <- as.data.frame(stri_list2matrix(OSvSS.ST.LT[["genelist_uu"]])) %>% rename("list1_uu" = 1, "list2_uu" = 2, "overlap_uu" = 3)
stlt.dd <- as.data.frame(stri_list2matrix(OSvSS.ST.LT[["genelist_dd"]])) %>% rename("list1_dd" = 1, "list2_dd" = 2, "overlap_dd" = 3)
stlt.du <- as.data.frame(stri_list2matrix(OSvSS.ST.LT[["genelist_du"]])) %>% rename("list1_du" = 1, "list2_du" = 2, "overlap_du" = 3)
stlt.ud <- as.data.frame(stri_list2matrix(OSvSS.ST.LT[["genelist_ud"]])) %>% rename("list1_ud" = 1, "list2_ud" = 2, "overlap_ud" = 3)


#save dfs
#OSvSS.paired.pvalue
write_csv(paired.uu, "OSvSS.paired.pvalue.RRHO.uu.genes.txt")
write_csv(paired.dd, "OSvSS.paired.pvalue.RRHO.dd.genes.txt")
write_csv(paired.ud, "OSvSS.paired.pvalue.RRHO.ud.genes.txt")
write_csv(paired.du, "OSvSS.paired.pvalue.RRHO.du.genes.txt")
#OSvSS.col.ST.pvalue
write_csv(st.uu, "OSvSS.col.ST.pvalue.RRHO.uu.genes.txt")
write_csv(st.dd, "OSvSS.col.ST.pvalue.RRHO.dd.genes.txt")
write_csv(st.ud, "OSvSS.col.ST.pvalue.RRHO.ud.genes.txt")
write_csv(st.du, "OSvSS.col.ST.pvalue.RRHO.du.genes.txt")
#OSvSS.col.LT.pvalue
write_csv(lt.uu, "OSvSS.col.LT.pvalue.RRHO.uu.genes.txt")
write_csv(lt.dd, "OSvSS.col.LT.pvalue.RRHO.dd.genes.txt")
write_csv(lt.ud, "OSvSS.col.LT.pvalue.RRHO.ud.genes.txt")
write_csv(lt.du, "OSvSS.col.LT.pvalue.RRHO.du.genes.txt")
#OSvSS.ST.LT.pvalue
write_csv(stlt.uu, "OSvSS.ST.LT.pvalue.RRHO.uu.genes.txt")
write_csv(stlt.dd, "OSvSS.ST.LT.pvalue.RRHO.dd.genes.txt")
write_csv(stlt.ud, "OSvSS.ST.LT.pvalue.RRHO.ud.genes.txt")
write_csv(stlt.du, "OSvSS.ST.LT.pvalue.RRHO.du.genes.txt")
```

make character vectors of overlapping genes from all quandrants
```{r}
#OSvSS.paired.pvalue
OSvSS.paired.UU <- OSvSS.paired.pvalue[["genelist_uu"]][["gene_list_overlap_uu"]]
OSvSS.paired.DD <- OSvSS.paired.pvalue[["genelist_dd"]][["gene_list_overlap_dd"]]
OSvSS.paired.UD <- OSvSS.paired.pvalue[["genelist_ud"]][["gene_list_overlap_ud"]]
OSvSS.paired.DU <- OSvSS.paired.pvalue[["genelist_du"]][["gene_list_overlap_du"]]

#collapsed PBond vs ST
colPB.ST.UU <- OSvSS.col.ST[["genelist_uu"]][["gene_list_overlap_uu"]]
colPB.ST.DD <- OSvSS.col.ST[["genelist_dd"]][["gene_list_overlap_dd"]]
colPB.ST.UD <- OSvSS.col.ST[["genelist_ud"]][["gene_list_overlap_ud"]]
colPB.ST.DU <- OSvSS.col.ST[["genelist_du"]][["gene_list_overlap_du"]]

#collapsed PBond vs LT
colPB.LT.UU <- OSvSS.col.LT[["genelist_uu"]][["gene_list_overlap_uu"]]
colPB.LT.DD <- OSvSS.col.LT[["genelist_dd"]][["gene_list_overlap_dd"]]
colPB.LT.UD <- OSvSS.col.LT[["genelist_ud"]][["gene_list_overlap_ud"]]
colPB.LT.DU <- OSvSS.col.LT[["genelist_du"]][["gene_list_overlap_du"]]

#ST vs LT
st.lt.UU <- OSvSS.ST.LT[["genelist_uu"]][["gene_list_overlap_uu"]]
st.lt.DD <- OSvSS.ST.LT[["genelist_dd"]][["gene_list_overlap_dd"]]
st.lt.UD <- OSvSS.ST.LT[["genelist_ud"]][["gene_list_overlap_ud"]]
st.lt.DU <- OSvSS.ST.LT[["genelist_du"]][["gene_list_overlap_du"]]

```

Fisher's exact test set up for Upset plots
```{r}
total = 12128
paletteFunc <- colorRampPalette(c('white', 'grey30'));
palette     <- paletteFunc(25)
pvalueShades = palette
```


Using Fisher's exact test to compare UU quads of all RRHOs
```{r}
upset <- list(OSvSS.paired.UU, colPB.ST.UU, colPB.LT.UU)
names(upset) <- c("48hrs vs 4wks Paired UU", "Collapsed PBond vs ST sep UU", "Collapsed PBond vs LT sep UU")
res = supertest(upset, n = total)

pdf("RRHO 48Pv4P compared to colST and colLT UU.pdf", width = 4, height = 4, onefile = TRUE)
plot(res, Layout="landscape", sort.by="size", margin=c(0.5,5,1,2), keep.empty.intersections = FALSE, show.expected.overlap=TRUE,expected.overlap.style="box", color.expected.overlap='black', color.on = "black", color.off = "whitesmoke", heatmapColor = pvalueShades, bar.split=c(2500, 3200))
dev.off()


#comparing with the ST vs LT as well
upset <- list(OSvSS.paired.UU, colPB.ST.UU, colPB.LT.UU, st.lt.UU)
names(upset) <- c("48hrs vs 4wks Paired UU", "Collapsed PBond vs ST sep UU", "Collapsed PBond vs LT sep UU", "48hrs vs 4wks Isolated UU")
res = supertest(upset, n = total)

pdf("RRHO 48Pv4P compared to colST and colLT and STLT UU.pdf", width = 8, height = 4, onefile = TRUE)
plot(res, Layout="landscape", sort.by="size", margin=c(0.5,5,1,2), keep.empty.intersections = FALSE, show.expected.overlap=TRUE,expected.overlap.style="box", color.expected.overlap='black', color.on = "black", color.off = "whitesmoke", heatmapColor = pvalueShades, bar.split=c(2500, 3200))
dev.off()
```

Using Fisher's exact test to compare DD quads of all RRHOs
```{r, fig.width= 8, fig.height= 7}
upset <- list(OSvSS.paired.DD, colPB.ST.DD, colPB.LT.DD)
names(upset) <- c("48hrs vs 4wks Paired DD", "Collapsed PBond vs ST sep DD", "Collapsed PBond vs LT sep DD")
res = supertest(upset, n = total)

#pdf("RRHO 48Pv4P compared to colST and colLT DD.pdf", width = 6, height = 5, onefile = TRUE)
plot(res, Layout="landscape", sort.by="size", margin=c(0.5,5,1,2), keep.empty.intersections = FALSE, show.expected.overlap=TRUE,expected.overlap.style="box", color.expected.overlap='black', color.on = "black", color.off = "whitesmoke", heatmapColor = pvalueShades, bar.split=c(2500, 3500))
#dev.off()



#comparing with the ST vs LT as well
upset <- list(OSvSS.paired.DD, colPB.ST.DD, colPB.LT.DD, st.lt.DD)
names(upset) <- c("48hrs vs 4wks Paired DD", "Collapsed PBond vs ST sep DD", "Collapsed PBond vs LT sep DD", "48hrs vs 4wks Isolated DD")
res = supertest(upset, n = total)

pdf("RRHO 48Pv4P compared to colST and colLT and STLT DD.pdf", width = 8, height = 5, onefile = TRUE)
plot(res, Layout="landscape", sort.by="size", margin=c(0.5,5,1,2), keep.empty.intersections = FALSE, show.expected.overlap=TRUE,expected.overlap.style="box", color.expected.overlap='black', color.on = "black", color.off = "whitesmoke", heatmapColor = pvalueShades, bar.split=c(2500, 3500))
dev.off()
```

make dfs of RRHO quadrants to prep for looking at specific gene overlap
```{r}
#OSvSS.paired
OSvSS.paired.UU.df <- as.data.frame(OSvSS.paired.UU) %>% dplyr::rename("genes" = 1)
OSvSS.paired.DD.df <- as.data.frame(OSvSS.paired.DD) %>% dplyr::rename("genes" = 1)
OSvSS.paired.UD.df <- as.data.frame(OSvSS.paired.UD) %>% dplyr::rename("genes" = 1)
OSvSS.paired.DU.df<- as.data.frame(OSvSS.paired.DU) %>% dplyr::rename("genes" = 1)

#collapsed PBond vs ST
colPB.ST.UU.df <- as.data.frame(colPB.ST.UU) %>% dplyr::rename("genes" = 1)
colPB.ST.DD.df <- as.data.frame(colPB.ST.DD) %>% dplyr::rename("genes" = 1)
colPB.ST.UD.df <- as.data.frame(colPB.ST.UD) %>% dplyr::rename("genes" = 1)
colPB.ST.DU.df <- as.data.frame(colPB.ST.DU) %>% dplyr::rename("genes" = 1)

#collapsed PBond vs LT
colPB.LT.UU.df <- as.data.frame(colPB.LT.UU) %>% dplyr::rename("genes" = 1)
colPB.LT.DD.df <- as.data.frame(colPB.LT.DD) %>% dplyr::rename("genes" = 1)
colPB.LT.UD.df <- as.data.frame(colPB.LT.UD) %>% dplyr::rename("genes" = 1)
colPB.LT.DU.df <- as.data.frame(colPB.LT.DU) %>% dplyr::rename("genes" = 1)

#ST vs LT 
STLT.UU.df <- as.data.frame(st.lt.UU) %>% dplyr::rename("genes" = 1)
STLT.DD.df <- as.data.frame(st.lt.DD) %>% dplyr::rename("genes" = 1)
STLT.UD.df <- as.data.frame(st.lt.UD) %>% dplyr::rename("genes" = 1)
STLT.DU.df <- as.data.frame(st.lt.DU) %>% dplyr::rename("genes" = 1)
```

gene lists for UU quads
```{r}
#common genes in UU quads of paired and colST and colLT RRHOs
common.all.uu <- OSvSS.paired.UU.df %>% filter(genes %in% colPB.ST.UU.df$genes) %>% filter(genes %in% colPB.LT.UU.df$genes) %>% rename("common.all.uu" = 1)

#common genes in UU quads of paired and colST RRHOs
common.toST.uu <- OSvSS.paired.UU.df %>% filter(genes %in% colPB.ST.UU.df$genes) %>% rename("common.paired.ST.uu" = 1)

#common genes in UU quads of paired and colLT RRHOs
common.toLT.uu <- OSvSS.paired.UU.df %>% filter(genes %in% colPB.LT.UU.df$genes) %>% rename("common.paired.LT.uu" = 1)

#common genes in UU quads of colST and colLT RRHOs
common.toST.LT.uu <- colPB.ST.UU.df %>% filter(genes %in% colPB.LT.UU.df$genes) %>% rename("common.ST.LT.uu" = 1)

#common genes in UU quads of colST and STLT RRHOs
common.toSTsep.bothseps.uu <- colPB.ST.UU.df %>% filter(genes %in% STLT.UU.df$genes) %>% rename("common.STsep.STLTsep.uu" = 1)

#common genes in UU quads of colLT and STLT RRHOs
common.toLTsep.bothseps.uu <- colPB.LT.UU.df %>% filter(genes %in% STLT.UU.df$genes) %>% rename("common.LTsep.STLTsep.uu" = 1)

#all of these dfs have the correct number of rows to match the upset plot
nrow(common.all.uu)
nrow(common.toST.uu)
nrow(common.toLT.uu)
nrow(common.toST.LT.uu)
nrow(common.toSTsep.bothseps.uu)
nrow(common.toLTsep.bothseps.uu)

#save dfs
```

gene lists for DD quads
```{r}
#common genes in DD quads of paired and colST and colLT RRHOs
common.all.dd <- OSvSS.paired.DD.df %>% filter(genes %in% colPB.ST.DD.df$genes)
common.all.dd <- common.all.dd %>% filter(genes %in% colPB.LT.DD.df$genes) %>% rename("common.all.dd" = 1)

#common genes in DD quads of paired and colST RRHOs
common.toST.dd <- OSvSS.paired.DD.df %>% filter(genes %in% colPB.ST.DD.df$genes) %>% rename("common.paired.ST.dd" = 1)

#common genes in DD quads of paired and colLT RRHOs
common.toLT.dd <- OSvSS.paired.DD.df %>% filter(genes %in% colPB.LT.DD.df$genes) %>% rename("common.paired.LT.dd" = 1)

#common genes in DD quads of colST and colLT RRHOs
common.toST.LT.dd <- colPB.ST.DD.df %>% filter(genes %in% colPB.LT.DD.df$genes) %>% rename("common.ST.LT.dd" = 1)

#comon genes in DD quads of colST and STLT RRHOs
common.toSTsep.bothseps.dd <- colPB.ST.DD.df %>% filter(genes %in% STLT.DD.df$genes) %>% rename("common.STsep.STLTsep.dd" = 1)

#comon genes in DD quads of colST and STLT RRHOs
common.toLTsep.bothseps.dd <- colPB.LT.DD.df %>% filter(genes %in% STLT.DD.df$genes) %>% rename("common.LTsep.STLTsep.dd" = 1)


#all of these dfs have the correct number of rows to match the upset plot
nrow(common.all.dd)
nrow(common.toST.dd)
nrow(common.toLT.dd)
nrow(common.toST.LT.dd)
nrow(common.toSTsep.bothseps.dd)
nrow(common.toLTsep.bothseps.dd)

#save dfs
```

finding differences in overalapping genes from UU quads
```{r}
#which genes are only found in the remain paired UU quad and not the ST or LT UU quads?
only.PBond.uu <- OSvSS.paired.UU.df %>% filter(!genes %in% colPB.ST.UU.df$genes)
only.PBond.uu <- only.PBond.uu %>% filter(!genes %in% colPB.LT.UU.df$genes)

#which genes are lost between ST and LT sep? 
diff.STtoLT.uu <- common.toST.uu %>% filter(!common.paired.ST.uu %in% common.toLT.uu$common.paired.LT.uu)

nrow(diff.STtoLT.uu)

int1 <- OSvSS.paired.UU.df %>% filter(genes %in% colPB.ST.UU.df$genes)
int2 <- OSvSS.paired.UU.df %>% filter(genes %in% colPB.LT.UU.df$genes)
int3 <- int1 %>% filter(!genes %in% int2$genes)
int4 <- int2 %>% filter(!genes %in% int1$genes)
diff <- int3 %>% filter(!genes %in% int4$genes)
diff2 <- int1 %>% filter(!genes %in% int3$genes) %>% filter(!genes %in% int4$genes)
nrow(int1)
nrow(int2)
nrow(int3)
nrow(int4)
nrow(diff)
nrow(diff2)

581 + 220
801 - 220
1701 - 1120
#save csv files
```

finding differences in overalapping genes from DD quads
```{r}
#which genes are only found in the remain paired DD quad and not the ST or LT DD quads?
only.PBond.dd <- OSvSS.paired.DD.df %>% filter(!genes %in% colPB.ST.DD.df$genes)
only.PBond.dd <- only.PBond.dd %>% filter(!genes %in% colPB.LT.DD.df$genes)

#which genes are lost between ST and LT sep? 
common.toST.dd <- common.toST.dd %>% rename("genes" = 1)
common.toLT.dd <- common.toLT.dd %>% rename("genes" = 1)
diff.STtoLT.dd <- common.toST.dd %>% filter(!genes %in% common.toLT.dd$genes)

test <- anti_join(common.toST.dd, common.toLT.dd, by = "genes")
#save csv files
```

difference in genes between ST and LT sep RRHOs
```{r}
diff.STtoLT.uu_direct <- colPB.ST.UU.df %>% filter(!genes %in% colPB.LT.UU.df$genes)
diff.STtoLT.dd_direct <- colPB.ST.DD.df %>% filter(!genes %in% colPB.LT.DD.df$genes)

nrow(diff.STtoLT.uu_direct)
nrow(diff.STtoLT.dd_direct)

#save csv files
```


GO analysis of intersection genes
rename column of all dfs needed for GO
```{r}
#don't need renaming
# OSvSS.paired.UU.df 
# OSvSS.paired.DD.df
# 
# colPB.ST.UU.df 
# colPB.ST.DD.df
# 
# colPB.LT.UU.df
# colPB.LT.DD.df

#rename for intersections of UU quads
common.all.uu <- common.all.uu %>% rename("genes" = 1)
common.toST.uu <- common.toST.uu %>% rename("genes" = 1)
common.toLT.uu <- common.toLT.uu %>% rename("genes" = 1)
common.toST.LT.uu <- common.toST.LT.uu %>% rename("genes" = 1)
common.toSTsep.bothseps.uu <- common.toSTsep.bothseps.uu %>% rename("genes" = 1)
common.toLTsep.bothseps.uu <- common.toLTsep.bothseps.uu %>% rename("genes" = 1)

#rename for intersetions of DD quads
common.all.dd <- common.all.dd %>% rename("genes" = 1)
common.toST.dd <- common.toST.dd %>% rename("genes" = 1)
common.toLT.dd <- common.toLT.dd %>% rename("genes" = 1)
common.toST.LT.dd <- common.toST.LT.dd %>% rename("genes" = 1)
common.toSTsep.bothseps.dd <- common.toSTsep.bothseps.dd %>% rename("genes" = 1)
common.toLTsep.bothseps.dd <- common.toLTsep.bothseps.dd %>% rename("genes" = 1)

#rename for differences in overlapping genes from UU quads
only.PBond.uu <- only.PBond.uu %>% rename("genes" = 1)
diff.STtoLT.uu <- diff.STtoLT.uu %>% rename("genes" = 1)
```

convert vole gene ids/ENSEMBL ids to mouse gene ids for all dfs for GO analysis
```{r}
#for UU quads 
forGO.OSvSS.paired.UU <- vole_to_mouse(OSvSS.paired.UU.df)
forGO.colPB.ST.UU <- vole_to_mouse(colPB.ST.UU.df)
forGO.colPB.LT.UU <- vole_to_mouse(colPB.LT.UU.df)
forGO.STLT.UU <- vole_to_mouse(STLT.UU.df)

#for DD quads
forGO.OSvSS.paired.DD <- vole_to_mouse(OSvSS.paired.DD.df)
forGO.colPB.ST.DD <- vole_to_mouse(colPB.ST.DD.df)
forGO.colPB.LT.DD <- vole_to_mouse(colPB.LT.DD.df)
forGO.STLT.DD <- vole_to_mouse(STLT.DD.df)

#for common genes of UU quads
forGO.common.all.uu <- vole_to_mouse(common.all.uu)
forGO.common.toST.uu <- vole_to_mouse(common.toST.uu)
forGO.common.toLT.uu <- vole_to_mouse(common.toLT.uu)
forGO.common.toST.LT.uu <- vole_to_mouse(common.toST.LT.uu)
forGO.common.PBST.STLT.uu <- vole_to_mouse(common.toSTsep.bothseps.uu)
forGO.common.PBLT.STLT.uu <- vole_to_mouse(common.toLTsep.bothseps.uu)

#for common genes of DD quads
forGO.common.all.dd <- vole_to_mouse(common.all.dd)
forGO.common.toST.dd <- vole_to_mouse(common.toST.dd)
forGO.common.toLT.dd <- vole_to_mouse(common.toLT.dd)
forGO.common.toST.LT.dd <- vole_to_mouse(common.toST.LT.dd)
forGO.common.PBST.STLT.dd <- vole_to_mouse(common.toSTsep.bothseps.dd)
forGO.common.PBLT.STLT.dd <- vole_to_mouse(common.toLTsep.bothseps.dd)

#for differences
forGO.only.PBond.uu <- vole_to_mouse(only.PBond.uu)
forGO.diff.STtoLT.uu <- vole_to_mouse(diff.STtoLT.uu)
```

make GO term dotplots for UU quads
```{r}
#I previously determined that Biological Processes is the best category for GO
OSvSS.paired.UU.plot.bp <- enrichGO(forGO.OSvSS.paired.UU$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "BP", pAdjustMethod = "fdr", readable = TRUE)
colPB.ST.UU.plot.bp <- enrichGO(forGO.colPB.ST.UU$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "BP", pAdjustMethod = "fdr", readable = TRUE)
colPB.LT.UU.plot.bp <- enrichGO(forGO.colPB.LT.UU$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "BP", pAdjustMethod = "fdr", readable = TRUE)
STLT.UU.plot.bp <- enrichGO(forGO.STLT.UU$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "BP", pAdjustMethod = "fdr", readable = TRUE)

#plot
barplot(OSvSS.paired.UU.plot.bp, showCategory = 20) 
barplot(colPB.ST.UU.plot.bp, showCategory = 20)
barplot(colPB.LT.UU.plot.bp, showCategory = 20)
barplot(STLT.UU.plot.bp, showCategory = 20)
```

make GO term dotplots for DD quads
```{r}
#I previously determined that Biological Processes is the best category for GO
OSvSS.paired.DD.plot.bp <- enrichGO(forGO.OSvSS.paired.DD$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "BP", pAdjustMethod = "fdr", readable = TRUE)
colPB.ST.DD.plot.bp <- enrichGO(forGO.colPB.ST.DD$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "BP", pAdjustMethod = "fdr", readable = TRUE)
colPB.LT.DD.plot.bp <- enrichGO(forGO.colPB.LT.DD$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "BP", pAdjustMethod = "fdr", readable = TRUE)
STLT.DD.plot.bp <- enrichGO(forGO.STLT.DD$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "BP", pAdjustMethod = "fdr", readable = TRUE)

#plot
barplot(OSvSS.paired.DD.plot.bp, showCategory = 20) 
barplot(colPB.ST.DD.plot.bp, showCategory = 20)
barplot(colPB.LT.DD.plot.bp, showCategory = 20)
barplot(STLT.DD.plot.bp, showCategory =  20)
```

I also want to know, what are the GO terms for the genes common to both colPB.ST and ST.LT? For UU and DD quadrants.
```{r}
colPBST.STLT.UU.plot.bp <- enrichGO(forGO.common.PBST.STLT.uu$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "BP", pAdjustMethod = "fdr", readable = TRUE)
colPBST.STLT.DD.plot.bp <- enrichGO(forGO.common.PBST.STLT.dd$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "BP", pAdjustMethod = "fdr", readable = TRUE)

colPBLT.STLT.UU.plot.bp <- enrichGO(forGO.common.PBLT.STLT.uu$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "BP", pAdjustMethod = "fdr", readable = TRUE)
colPBLT.STLT.DD.plot.bp <- enrichGO(forGO.common.PBLT.STLT.dd$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "BP", pAdjustMethod = "fdr", readable = TRUE)

#plot
barplot(colPBST.STLT.UU.plot.bp, showCategory =  20)
barplot(colPBST.STLT.DD.plot.bp, showCategory =  20)
barplot(colPBLT.STLT.UU.plot.bp, showCategory =  20)
barplot(colPBLT.STLT.DD.plot.bp, showCategory =  20)
```

