---
title: "Sankey plot of clustered bulk and neuronal intersection colPBond_STsep_LTsep"
output:
  pdf_document: default
  html_notebook: default
---

This is an [R Markdown](http://rmarkdown.rstudio.com) Notebook. When you execute code within the notebook, the results appear beneath the code. 

load libraries 
```{r}
library(tidyverse)
library(networkD3)
```

data I used to generate the clustered heatmaps 
```{r}
#For bulk RNAseq {dir}/2021_05_13_morpheus/morpheus_colPBvsSeparations
#The neuronal intersection gene list was generated by which of the significanlty, positively correlated modules from WGCNA were found in the DEGs of the TRAP Inputs vs Pulldowns. This info is in the notebook entitled "2021_06_06_neuronalgenesOnly_morpheus" and the data is in the file: {dir}/WGCNA/Intersection of all pos corr WGCNA modules and TRAP IvP pulldowns.txt
    #I then used the "neuronal intersection genes" to filter the bulk RNAseq of line 15. I put both of these into Morpheus separately and clustered using One Minus Spearman's correlation using complete clustering. 
#I then manually annotated the clusters by color and am importing that data where the gene is attributed to a cluster for both bulk and neuronal clusters. I now want to know where the neuronal genes from each cluster are in the bulk clusters. This will be visualized by a sankey plot. 
```

import data for genes with cluster designations
```{r}
bulk_clusters <- read_excel("{dir}/clustered DESeq of bulk and neuronal colPBond_STsep_LTsep.xlsx", sheet = "bulk clusters edited")
neuronal_clusters <- read_excel("{dir}/clustered DESeq of bulk and neuronal colPBond_STsep_LTsep.xlsx", sheet = "neuronal only clusters edited")
```

how many genes are in each cluster for the bulk 
```{r}
sum_bulkClusters <- table(unlist(bulk_clusters$cluster_color))
```


```{r}
sum_red <- sankey_overlap(neuronal_clusters, bulk_clusters, clusterColor = "red")
sum_orange <- sankey_overlap(neuronal_clusters, bulk_clusters, clusterColor = "orange")
sum_yellow <- sankey_overlap(neuronal_clusters, bulk_clusters, clusterColor = "yellow")
sum_green <- sankey_overlap(neuronal_clusters, bulk_clusters, clusterColor = "green")
sum_blue <- sankey_overlap(neuronal_clusters, bulk_clusters, clusterColor = "blue")
sum_purple <- sankey_overlap(neuronal_clusters, bulk_clusters, clusterColor = "purple")
sum_periwinkle <- sankey_overlap(neuronal_clusters, bulk_clusters, clusterColor = "periwinkle")
sum_pink <- sankey_overlap(neuronal_clusters, bulk_clusters, clusterColor = "pink")
sum_teal <- sankey_overlap(neuronal_clusters, bulk_clusters, clusterColor = "teal")
sum_limegreen <- sankey_overlap(neuronal_clusters, bulk_clusters, clusterColor = "limegreen")
sum_babyblue <- sankey_overlap(neuronal_clusters, bulk_clusters, clusterColor = "babyblue")
```
I consolidated all of this data into a format for Sankey in an excel sheet 

import sankey plot data
```{r}
sankey_data <- read_excel("{dir}/clustered DESeq of bulk and neuronal colPBond_STsep_LTsep.xlsx", sheet = "try_again")

nodes <- data.frame(name = c(as.character(sankey_data$source), as.character(sankey_data$target)) %>% unique())

sankey_data$IDsource <- match(sankey_data$source, nodes$name)-1
sankey_data$IDtarget <- match(sankey_data$target, nodes$name)-1
sankey_data$group <- as.factor(c("i", "ni", "ni", "ni", "ni", "ni", "ni", "ni", "ni", "ni", "ni", "ni", "ni", "ni", "i", "i", "ni", "ni", "ni", "i", "i"))
nodes$group <- as.factor(c("1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "2", "7", "8", "1", "4", "3", "5", "6", "9", "10"))
```


ok let's make this look nicer
```{r}
my_color <- 'd3.scaleOrdinal() .domain([ "i", "ni", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "2", "7", "8", "1", "4", "3", "5", "6", "9", "10"]) .range(["#9a8f97", "#e9e3e6", "#54478c", "#2c699a", "#048ba8", "#0db39e", "#16db93", "#83e377", "#b9e769", "#efea5a", "#f1c453", "#f29e4c", "#D26D0F", "#2c699a", "#b9e769", "#efea5a", "#54478c", "#0db39e", "#048ba8", "#16db93", "#83e377", "#f1c453", "#f29e4c"])'

sankeyNetwork(Links = sankey_data, Nodes = nodes, Source = "IDsource", Target = "IDtarget", Value = "value", NodeID = "name", sinksRight = FALSE, colourScale = my_color, LinkGroup = "group", NodeGroup = "group", nodePadding = 0)

webshot("file:///{dir}/Figures/2021_06_16_sankeyPlot/with_colors_ordered.html", "colored_sankeyplot.pdf", delay = 0.2)
```

Now looking at GO terms for the neuronal clusters of interest and the bulk clusters they map to
import mouse genome for converting ids
```{r}
mart_export_PVEnsembl_mouseEnsembl_mousegeneName <- read.csv("{dir}/mart_export_PVEnsembl_mouseEnsembl_mousegeneName.txt", stringsAsFactors=FALSE)

#getting a biomart df that only contains rows that have mouse stable IDs 
biomart_mouseE <- filter(mart_export_PVEnsembl_mouseEnsembl_mousegeneName, trimws(Mouse.gene.stable.ID) !="")
```

First I'm looking at neuronal cluster "red" which maps to bulk cluster "orange"
```{r}
bulk_orange <- bulk_clusters %>% filter(cluster_color == "orange")
neuronal_red <- neuronal_clusters %>% filter(cluster_color == "red")

#covert vole to mouse gene names
go.bulk_orange <- vole_to_mouse(bulk_orange)
go.neuronal_red <- vole_to_mouse(neuronal_red)
```

run GO 
```{r}
go.bulkOrange.plot.bp <- enrichGO(go.bulk_orange$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "BP", pAdjustMethod = "fdr", readable = TRUE)
go.neunRed.plot.bp <- enrichGO(go.neuronal_red$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "BP", pAdjustMethod = "fdr", readable = TRUE)

barplot(go.bulkOrange.plot.bp)
barplot(go.neunRed.plot.bp)
#hmm ok since the list of neunRed is so short (99 genes), there aren't any terms that pass significance. BUT the orange cluster has lots of glial stuff
```

Now I'm looking at neuronal cluster "pink" which maps to bulk clusters "pink" and "teal"
```{r}
bulk_pinkTeal <- bulk_clusters %>% filter(cluster_color == "pink" | cluster_color == "teal")
neuronal_pink <- neuronal_clusters %>% filter(cluster_color == "pink")

#covert vole to mouse gene names
go.bulk_pinkTeal <- vole_to_mouse(bulk_pinkTeal)
go.neuronal_pink <- vole_to_mouse(neuronal_pink)

#run GO 
go.bulkPinkTeal.plot.bp <- enrichGO(go.bulk_pinkTeal$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "BP", pAdjustMethod = "fdr", readable = TRUE)
go.neunPink.plot.bp <- enrichGO(go.neuronal_pink$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "BP", pAdjustMethod = "fdr", readable = TRUE)

barplot(go.bulkPinkTeal.plot.bp)
barplot(go.neunPink.plot.bp)

#Terms like "response to amphetamine" and "response to amine" are really similar but include DRD1A and DRD2. These genes don't come up in bulk terms until term 34. So there's a huge enrichment of identifying DRD1A and DRD2 in the neuronal only genes. 
#In bulk data, DRD1A and DRD2 are in different clusters. DRD2 is in "teal" and DRD1A is in "pink". while in the neuronal data both are in "pink." 
```

Finally, I'm looking at neuronal cluster "babyblue" which maps to bulk clusters "purple" and "limegreen"
```{r}
bulk_purpleLime <- bulk_clusters %>% filter(cluster_color == "purple" | cluster_color == "limegreen")
neuronal_babyblue <- neuronal_clusters %>% filter(cluster_color == "babyblue")

#covert vole to mouse gene names
go.bulk_purpleLime <- vole_to_mouse(bulk_purpleLime)
go.neuronal_babyblue <- vole_to_mouse(neuronal_babyblue)

#run GO 
go.bulkPurpleLime.plot.bp <- enrichGO(go.bulk_purpleLime$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "BP", pAdjustMethod = "fdr", readable = TRUE)
go.neunBabyblue.plot.bp <- enrichGO(go.neuronal_babyblue$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "BP", pAdjustMethod = "fdr", readable = TRUE)

barplot(go.bulkPurpleLime.plot.bp)
barplot(go.neunBabyblue.plot.bp)
```

Now let's run GO on the neuronal only genes
```{r}
go.neuronal <- vole_to_mouse(neuronal_clusters)
go.neuronal.plot.bp <- enrichGO(go.neuronal$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "BP", pAdjustMethod = "fdr", readable = TRUE)
go.neuronal.plot.cc <- enrichGO(go.neuronal$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "CC", pAdjustMethod = "fdr", readable = TRUE)
go.neuronal.plot.all <- enrichGO(go.neuronal$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "ALL", pAdjustMethod = "fdr", readable = TRUE)

barplot(go.neuronal.plot.bp)
barplot(go.neuronal.plot.cc)
barplot(go.neuronal.plot.all)
#GABA receptors are highly represented 

go.bulk <- vole_to_mouse(bulk_clusters)
go.bulk.plot.bp <- enrichGO(go.bulk$ENTREZID, OrgDb = org.Mm.eg.db, pool = TRUE, ont= "BP", pAdjustMethod = "fdr", readable = TRUE)
barplot(go.bulk.plot.bp)
#GABA receptor terms aren't AS highly represented but still like the 5th term
```

make filtered GO plots in preperation for Morpheus
```{r}
#first get GO IDs from each neuronal cluster of interest
neunRed_IDs <- as.data.frame(go.neunRed.plot.bp@result$ID)
neunRed_IDs <- neunRed_IDs[c(1, 3, 5, 6, 7, 9), ]

neunPink_IDs <- as.data.frame(go.neunPink.plot.bp@result$ID)
neunPink_IDs <- neunPink_IDs[c(1, 6, 7, 13, 14, 15), ]

neunBB_IDs <- as.data.frame(go.neunBabyblue.plot.bp@result$ID)
neunBB_IDs <- neunBB_IDs[c(1, 2, 3, 6, 7, 8), ]

#now filter the bulk GO dfs by the appropriate IDs
filtered_bulkOrange_go <- go.bulkOrange.plot.bp@result %>% filter(ID %in% neunRed_IDs) %>% select(c(2,6)) %>% mutate(p.adjust = -log10(p.adjust)) %>% rename("bulkOrange_padjust" = 2)
filtered_pinkteal_go <- go.bulkPinkTeal.plot.bp@result %>% filter(ID %in% neunPink_IDs) %>% select(c(2,6)) %>% mutate(p.adjust = -log10(p.adjust)) %>% rename("bulkPinkTeal_padjust" = 2)
filtered_purpleLime_go <- go.bulkPurpleLime.plot.bp@result %>% filter(ID %in% neunBB_IDs) %>% select(c(2,6)) %>% mutate(p.adjust = -log10(p.adjust)) %>% rename("bulkPurpleLime_padjust" = 2)

#filter the neuronal GO dfs by the appropriate IDs
filtered_neunRed_go <- go.neunRed.plot.bp@result %>% slice(c(1, 3, 5, 6, 7, 9)) %>% select(c(2,6)) %>% mutate(p.adjust = -log10(p.adjust)) %>% rename("neunRed_padjust" = 2)
filtered_neunPink_go <- go.neunPink.plot.bp@result %>% slice(c(1, 6, 7, 13, 14, 15)) %>% select(c(2,6)) %>% mutate(p.adjust = -log10(p.adjust)) %>% rename("neunPink_padjust" = 2)
filtered_neunBB_go <- go.neunBabyblue.plot.bp@result %>% slice(c(1, 2, 3, 6, 7, 8)) %>% select(c(2,6)) %>% mutate(p.adjust = -log10(p.adjust)) %>% rename("neunBB_padjust" = 2)
```

now put together the appropriate comparisons and save for Morpheus
```{r}
NRed_to_BOrange <- left_join(filtered_neunRed_go, filtered_bulkOrange_go, by = "Description")
NPink_to_BPTeal <- left_join(filtered_neunPink_go, filtered_pinkteal_go, by = "Description")
NBB_to_BPLime <- left_join(filtered_neunBB_go, filtered_purpleLime_go, by = "Description")

setwd("{dir}/DEGs/2021_06_21_TRAPmorpheus")
write_csv(NRed_to_BOrange, "GO terms of neuronal cluster red to bulk orange.txt")
write_csv(NPink_to_BPTeal, "GO terms of neuronal cluster pink to bulk PinkTeal.txt")
write_csv(NBB_to_BPLime, "GO terms of neuronal cluster babyblue to bulk PurpleLime.txt")
```


